### ISSUE TITLE
Make the EntityAboutCard fields customisable

### ISSUE THREAD

**Issue by freben** [backstage/backstage#6175]
## Feature Suggestion

The `EntityAboutCard` from `@backstage/plugin-catalog` shows a fixed set of fields. It could be useful to be able to easily replace or amend that set of fields with custom ones, reusing the rest of the card as-is.

## Possible Implementation

The field component is already exported as such, but the card does not accept a custom list of fields yet. The card could be made to accept an optional children prop, `children?: React.ReactNode`. If it is undefined, it renders the current list of default fields. If it's defined, it renders only those fields.

This `children` prop will be passed down into `AboutContent` which is responsible for drawing them.

This would probably best be accompanied with having easy access to a set of default fields. See what the entity table does [here](https://github.com/backstage/backstage/blob/master/plugins/catalog-react/src/components/EntityTable/EntityTable.tsx#L77) for inspiration. Maybe there would be `EntityAboutTable.fields` which is all of the available default fields in an object, and `AboutTable.defaultFields` as an array, so you can do

```ts
<EntityAboutCard>
  {...EntityAboutCard.defaultFields}
  <MyCustomAboutField1 />
</EntityAboutCard>
```

or

```ts
<EntityAboutCard>
  <EntityAboutCard.fields.Description />
  <MyCustomAboutField1 />
</EntityAboutCard>
```

where `MyCustomAboutField1` can use the `useEntity()` hook to get access to the actual entity that is being displayed.

## Context

https://github.com/backstage/backstage/issues/6164#issuecomment-866554734

**Comment by Seneli** [comment]
Hi, I would like to try this one.

**Comment by benjdlambert** [comment]
Assigned to you @Seneli!

**Comment by chuliang014** [comment]
Any update on this?

**Comment by Seneli** [comment]
Hi @chuliang014, I've been working on it the last few weeks but got stuck on one area. 

Currently what I'm trying to do is assume the custom fields get passed in as an array of objects to the `AboutCard` component via an optional children prop, similar to what freben suggested, but I'm struggling to test it out using the `AboutCard.test.tsx` file, since I'm new to Jest. I think once this is done I can push the change and request a peer review. I just don't want to push something I haven't properly tested yet.  

What I currently have set up is that the children prop gets passed into `AboutCard` and into the child component `AboutContent`. If the child component is undefined, the card displays the default fields, but if it is defined it displays the "description" and "tags" default fields with all the custom fields in the middle. All fields are displayed using the `AboutField` component.

**Comment by chuliang014** [comment]
Great! At the moment,  I just copied the entire the component of AboutCard for our use.

**Comment by Seneli** [comment]
What do you mean by "copied the entire component of AboutCard"? Is there something I should watch out for?

**Comment by freben** [comment]
@Seneli I presume they meant taking the [source code](https://github.com/backstage/backstage/tree/master/plugins/catalog/src/components/AboutCard) for the About card and lifting that over to their local project.

**Comment by pallavipervaje** [comment]
@Seneli @freben Can we customize EntityAboutCard fields now?

**Comment by jpeach** [comment]
See also https://github.com/backstage/backstage/issues/5813

**Comment by namco1992** [comment]
@freben @Seneli Hi 👋 are we still keen on implementing this? Will this be parked until https://github.com/backstage/backstage/issues/11035 or https://github.com/backstage/backstage/pull/11819 is settled? I would like to take on what's left if possible, thanks!

**Comment by freben** [comment]
I don't think is a case for https://github.com/backstage/backstage/issues/11035 or https://github.com/backstage/backstage/pull/11819 . This issue is basically up for grabs, I think, for someone who wants to contribute.

**Comment by freben** [comment]
Unassigned @Seneli for now. Let me know if you (or someone else!) wants to pick up the baton.

**Comment by namco1992** [comment]
Thanks @freben! Yes pls assign it to me and I will work on it.

**Comment by Rugvip** [comment]
Closing this until we have a better idea of how it could be done
